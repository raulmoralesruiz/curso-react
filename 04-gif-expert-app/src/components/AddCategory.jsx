import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {

  const [inputValue, setInputValue] = useState('');

  const onInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const onFormSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      onNewCategory(inputValue);
      setInputValue("");  
    }
  };

  return (
    <form onSubmit={ onFormSubmit }>
      
      <input
        type="text"
        placeholder="Buscar..."
        value={inputValue}
        onChange={onInputChange}
      />

    </form>

  );
};
