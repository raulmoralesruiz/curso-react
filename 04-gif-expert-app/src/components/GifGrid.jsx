import { getGifts } from "../helpers/getGifs";

export const GifGrid = ({ category }) => {

  getGifts(category);

  return (
    <>
      <h3>{ category }</h3>
    </>
  )
}
