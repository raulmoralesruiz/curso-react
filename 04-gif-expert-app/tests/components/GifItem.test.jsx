import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem.jsx />', () => {

  const title = 'Un título';
  const url = 'https://aboutme.raulmorales.eu/';

  test('Debe hacer match con el snapshot', () => {
    const {container} = render( <GifItem title={title} url={url} /> );
    expect(container).toMatchSnapshot();
  });

  test('Debe mostrar la imagen con el URL y ALT indicado', () => {
    render( <GifItem title={title} url={url} /> );
    // screen.debug();
    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title);
    const {src, alt} = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Debe mostrar el título en el componente', () => {
    render( <GifItem title={title} url={url} /> );
    expect(screen.getByText(title)).toBeTruthy();
  });
  
})