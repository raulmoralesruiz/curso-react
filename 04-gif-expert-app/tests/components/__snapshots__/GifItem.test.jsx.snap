// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem.jsx /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Un título"
      src="https://aboutme.raulmorales.eu/"
    />
    <p>
      Un título
    </p>
  </div>
</div>
`;
